src/utilities/viewer.cpp

```diff
@@ -990,7 +1003,28 @@ LRESULT CALLBACK TestViewerWindow::WindowProc(
     case WM_KEYUP:
       gInstance->OnKeyUp(wParam);
       return DefWindowProc(hWnd, uMsg, wParam, lParam);
+    case WM_SIZE:
+      [[fallthrough]];
+    case WM_MOVE: {
+      if (!IsIconic(hWnd)) {
+        RECT windowRect;
+        if (GetWindowRect(hWnd, &windowRect)) {
+          gInstance->mWindowRect = windowRect;
+        }
+      }
+      return 0;
+    }
     case WM_CLOSE:
+
+      gInstance->mSettings.mWindowX = gInstance->mWindowRect.left;
+      gInstance->mSettings.mWindowY = gInstance->mWindowRect.top;
+
+      gInstance->mSettings.mWindowWidth
+        = gInstance->mWindowRect.right - gInstance->mWindowRect.left;
+      gInstance->mSettings.mWindowHeight
+        = gInstance->mWindowRect.bottom - gInstance->mWindowRect.top;
+
+      gInstance->mSettings.Save();

```

Comment:
This feels like enough member accesses that it might be worth moving to a new method